src/index.ts

```diff
@@ -1,8 +1,53 @@
-import type { TransformedSource, Transformer, TransformOptions } from '@jest/transform';
+import type { CacheKeyOptions, TransformedSource, Transformer, TransformOptions } from '@jest/transform';
 import type { Config } from '@jest/types';
+import createCacheKey from '@jest/create-cache-key-function';
 import { TsJestTransformer } from 'ts-jest/dist/ts-jest-transformer';
 
+import { NgJestConfig } from './config/ng-jest-config';
+import { stringify } from './utils/json';
+import { JsonableValue } from './utils/jsonable-value';
+
+interface ConfigSetItem {
+  configSet: NgJestConfig;
+  jestConfig: JsonableValue<Config.ProjectConfig>;
+}
+
 class AngularJestTransformer extends TsJestTransformer implements Transformer {
+  /**
+   * @internal
+   * cache config set between each test run
+   */
+  private static readonly configSetItems: ConfigSetItem[] = [];
+
+  /**
+   * Override `ts-jest` configsFor to initialize our `NgJestConfig`
+   * @internal
+   */
+  configsFor(jestConfig: Config.ProjectConfig): NgJestConfig {
+    let csi: ConfigSetItem | undefined = AngularJestTransformer.configSetItems.find(
+      (cs) => cs.jestConfig.value === jestConfig,
+    );
+    if (csi) return csi.configSet;
+    // try to look-it up by stringified version
+    const serialized = stringify(jestConfig);
+    csi = AngularJestTransformer.configSetItems.find((cs) => cs.jestConfig.serialized === serialized);
+    if (csi) {
+      // update the object so that we can find it later
+      // this happens because jest first calls getCacheKey with stringified version of
+      // the config, and then it calls the transformer with the proper object
```

Comment:
I think I need more information about this edge case :smile:

Reply:
This one handles the case that when cannot find any jest config by memory reference in `configSetItems`, we fallback to find by serialized string. If match, we simply update memory reference. This case hardly happens, just for safety.